docs/arguments.rst

```diff
@@ -34,166 +37,86 @@ And what it looks like:
-    def copy(src, dst):
+    @click.argument('src', nargs=1)
+    @click.argument('dsts', nargs=-1)
+    def copy(src, dsts):
         """Move file SRC to DST."""
-        for fn in src:
-            click.echo(f"move {fn} to folder {dst}")
+        for destination in dsts:
+            click.echo(f"Copy {src} to folder {destination}")
 
 And what it looks like:
 
 .. click:run::
 
-    invoke(copy, args=['foo.txt', 'bar.txt', 'my_folder'])
-
-Note that this is not how you would write this application.  The reason
-for this is that in this particular example the arguments are defined as
-strings.  Filenames, however, are not strings!  They might be on certain
-operating systems, but not necessarily on all.  For better ways to write
-this, see the next sections.
-
-.. admonition:: Note on Non-Empty Variadic Arguments
-
-   If you come from ``argparse``, you might be missing support for setting
-   ``nargs`` to ``+`` to indicate that at least one argument is required.
+    invoke(copy, args=['foo.txt', 'usr/david/foo.txt', 'usr/mitsuko/foo.txt'])
 
-   This is supported by setting ``required=True``.  However, this should
-   not be used if you can avoid it as we believe scripts should gracefully
-   degrade into becoming noops if a variadic argument is empty.  The
-   reason for this is that very often, scripts are invoked with wildcard
-   inputs from the command line and they should not error out if the
-   wildcard is empty.
+.. admonition:: Note on Handling Files.
 
-.. _file-args:
+    This is not how you would handle files and files paths, since it passes them in as strings. For more see :ref:`handling-files`.
 
-File Arguments
---------------
+Argument Escape Sequences
+---------------------------
 
-Since all the examples have already worked with filenames, it makes sense
-to explain how to deal with files properly.  Command line tools are more
-fun if they work with files the Unix way, which is to accept ``-`` as a
-special file that refers to stdin/stdout.
+If you want to process arguments that look like options, like a file named ``-foo.txt`` or ``--foo.txt`` , you must pass the ``--`` separator first. After you pass the ``--``, you may only pass arguments. This is a common feature for POSIX command line tools.
 
-Click supports this through the :class:`click.File` type which
-intelligently handles files for you.  It also deals with Unicode and bytes
-correctly for all versions of Python so your script stays very portable.
-
-Example:
+Example usage:
 
 .. click:example::
 
     @click.command()
-    @click.argument('input', type=click.File('rb'))
-    @click.argument('output', type=click.File('wb'))
-    def inout(input, output):
-        """Copy contents of INPUT to OUTPUT."""
-        while True:
-            chunk = input.read(1024)
-            if not chunk:
-                break
-            output.write(chunk)
-
-And what it does:
-
-.. click:run::
+    @click.argument('files', nargs=-1, type=click.Path())
+    def touch(files):
```

Comment:
```suggestion
    @click.argument('files', nargs=-1, type=click.Path(path_type=pathlib.Path))
    def touch(files: tuple[pathlib.Path, ...]):
```

I feel like we should make `path_type=pathlib.Path` the default. Until it's not, let's maybe make it a bit more clear they can get a `Path` type here as it otherwise defaults to `str`.